File: Medium/1448 Count Good Nodes in Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    good=0
    def goodNodes(self, root: TreeNode) -> int:
        def dfs(node,val):
            if not node: return 
            if node.val>=val:
                self.good+=1
                val=node.val
            dfs(node.left,val)
            dfs(node.right,val)
        dfs(root,root.val)
        return self.good

File: Medium/1448 Count Good Nodes in Binary Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int good=0;
    public void dfs(TreeNode node, int val){
        if(node==null) return;
        if(node.val>=val){
            good++;
            val=node.val;
        }
        dfs(node.left,val);
        dfs(node.right,val);
    }
    public int goodNodes(TreeNode root) {
        dfs(root,root.val);
        return good;
    }
}
